crates/bevy_reflect/src/list.rs

```diff
@@ -9,18 +9,41 @@ use crate::{
 
 /// An ordered, mutable list of [Reflect] items. This corresponds to types like [`std::vec::Vec`].
 ///
-/// This is a sub-trait of [`Array`] as it implements a [`push`](List::push) function, allowing
-/// it's internal size to grow.
+/// This is a sub-trait of [`Array`] as it implements [insertion](List::insert) and [removal](List::remove),
+/// allowing it's internal size to grow.
 ///
 /// This trait expects index 0 to contain the _front_ element.
 /// The _back_ element must refer to the element with the largest index.
 /// These two rules above should be upheld by manual implementors.
+///
+/// [`push`](List::push) and [`pop`](List::pop) have default implementations,
+/// however it may be faster to implement them manually.
 pub trait List: Reflect + Array {
+    /// Inserts an element at position `index` within the list.
+    ///
+    /// # Panics
+    /// Panics if `index > len`.
+    fn insert(&mut self, index: usize, element: Box<dyn Reflect>);
+
+    /// Removes and returns the element at position `index` within the list.
```

Comment:
Nit: Similarly, should we update these docs to [match](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.remove) as well?
```suggestion
    /// Removes and returns the element at position `index` within the list, shifting all elements after it to the left.
```

Reply:
```suggestion
    /// Removes and returns an element at position `index` within the list,
    /// shifting all elements after it towards the front of the list.
```